> @chentong7 does `npm run eslint` (in getkeys) finish successfully for you? I'm seeing this error:
> 
> ```shell
> $ npm run eslint
> 
> > @fluid-internal/getkeys@1.0.0 eslint /workspaces/FluidFramework/tools/getkeys
> > eslint ./index.js
> 
> 
> Oops! Something went wrong! :(
> 
> ESLint: 8.6.0
> 
> Error: Failed to load plugin '@typescript-eslint' declared in '.eslintrc.js » /workspaces/FluidFramework/tools/getkeys/node_modules/@fluidframework/eslint-config-fluid/minimal.js » ./eslint7': Cannot find module 'typescript'
> Require stack:
> - /workspaces/FluidFramework/tools/getkeys/node_modules/@typescript-eslint/eslint-plugin/dist/util/astUtils.js
> - /workspaces/FluidFramework/tools/getkeys/node_modules/@typescript-eslint/eslint-plugin/dist/util/index.js
> - /workspaces/FluidFramework/tools/getkeys/node_modules/@typescript-eslint/eslint-plugin/dist/rules/adjacent-overload-signatures.js
> - /workspaces/FluidFramework/tools/getkeys/node_modules/@typescript-eslint/eslint-plugin/dist/rules/index.js
> - /workspaces/FluidFramework/tools/getkeys/node_modules/@typescript-eslint/eslint-plugin/dist/index.js
> - /workspaces/FluidFramework/tools/getkeys/node_modules/@eslint/eslintrc/dist/eslintrc.cjs
> Referenced from: /workspaces/FluidFramework/tools/getkeys/node_modules/@fluidframework/eslint-config-fluid/eslint7.js
>     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
>     at Function.Module._load (internal/modules/cjs/loader.js:746:27)
>     at Module.require (internal/modules/cjs/loader.js:974:19)
>     at require (/workspaces/FluidFramework/tools/getkeys/node_modules/v8-compile-cache/v8-compile-cache.js:159:20)
>     at Object.<anonymous> (/workspaces/FluidFramework/tools/getkeys/node_modules/@typescript-eslint/eslint-plugin/dist/util/astUtils.js:27:25)
>     at Module._compile (/workspaces/FluidFramework/tools/getkeys/node_modules/v8-compile-cache/v8-compile-cache.js:192:30)
>     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
>     at Module.load (internal/modules/cjs/loader.js:950:32)
>     at Function.Module._load (internal/modules/cjs/loader.js:790:12)
>     at Module.require (internal/modules/cjs/loader.js:974:19)
> npm ERR! code ELIFECYCLE
> npm ERR! errno 2
> npm ERR! @fluid-internal/getkeys@1.0.0 eslint: `eslint ./index.js`
> npm ERR! Exit status 2
> npm ERR! 
> npm ERR! Failed at the @fluid-internal/getkeys@1.0.0 eslint script.
> npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /home/node/.npm/_logs/2022-10-25T15_46_58_274Z-debug.log
> ```
> 
> If this is just something I missed in my env, ignore it. But if it reproduces for you, we might need a dev-dependency on `"typescript": "~4.5.5"` to address it? Which as @Josmithr pointed out, ideally we shouldn't need because this package does not have TypeScript code, but if our current lint configs assume it does...
> 
> If we end up going that route, it's probably worth it to document in this package's readme all the things we had to do as a workaround to make our lint config happy (which dependencies we added to `package.json`, which rules we put in `.eslintrc.js` that shouldn't be necessary, etc), so they can easily be undone when we have a more correct base lint config.
> 
> Also, I think we can close #12506 ? Or do we need it open for some reason?

Yes, `npm run lint` or `npm run eslint` runs fine on my end. I think you probably need to remove node_module directory and re-install? And also I did the npm -g install eslint in my local environment.

--- a/tools/getkeys/.eslintrc.cjs
+++ b/tools/getkeys/.eslintrc.cjs
@@ -1,0 +1,31 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+module.exports = {
+    "extends": [
+        require.resolve("@fluidframework/eslint-config-fluid/minimal")
+    ],
+    "parserOptions": {
+        project: ["./tsconfig.json"],
+    },
+    "settings": {
+        "import/resolver": "node"
+    },
+    "rules": {
+        /**
+         * TODO: no-unsafe-* ts rules should be excluded for .js files
+         */
+        "@typescript-eslint/ban-ts-comment": "off",
+        "@typescript-eslint/no-non-null-assertion": "off",
+        "@typescript-eslint/no-unsafe-argument": "off",
+        "@typescript-eslint/no-unsafe-assignment": "off",
+        "@typescript-eslint/no-unsafe-call":"off",
+        "@typescript-eslint/no-unsafe-member-access": "off",
+        "@typescript-eslint/restrict-plus-operands": "off",
+        "@typescript-eslint/strict-boolean-expressions": "off",
+        "import/no-nodejs-modules": "off",
+        "promise/param-names": "off"
+    }
+}

--- a/tools/getkeys/index.js
+++ b/tools/getkeys/index.js
@@ -3,14 +3,14 @@
  * Licensed under the MIT License.
  */
 
-const fs = require("fs");
-const os = require("os");
-const path = require("path");
-const util = require("util");
-const { exec } = require('child_process');
-const msRestAzure = require("ms-rest-azure");
-const keyVault = require("azure-keyvault");
-const rcTools = require("@fluidframework/tool-utils");
+import fs from "fs";
+import os from "os";
+import path from "path";
+import util from "util";
+import child_process from "child_process";
+import interactiveLogin from "ms-rest-azure";
+import KeyVaultClient from "azure-keyvault";
+import { loadRC, saveRC } from "@fluidframework/tool-utils";
 
 const appendFile = util.promisify(fs.appendFile);
 
@@ -30,6 +30,7 @@ async function exportToShellRc(shellRc, entries) {
         entries.map(([key, value]) => `export ${key}=${quote(value)}`).join("\n")
     }\n`;
 
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
     return appendFile(rcPath, stmts, "utf-8");
 }
 
@@ -61,7 +62,7 @@ async function saveEnv(env) {
             // For 'fish' we use 'set -xU', which dedupes and performs its own escaping.
             // Note that we must pass the 'shell' option, otherwise node will spawn '/bin/sh'.
             return Promise.all(
-                entries.map(([key, value]) => execAsync(`set -xU '${key}' '${value}'`, { shell }))
+                entries.map(async ([key, value]) => execAsync(`set -xU '${key}' '${value}'`, { shell })),
             );
         default:
             if (!process.platform === "win32") {
@@ -71,7 +72,7 @@ async function saveEnv(env) {
 
                 // On Windows, invoke 'setx' to update the user's persistent environment variables.
                 return Promise.all(
-                    entries.map(([key, value]) => execAsync(`setx ${key} ${quote(value)}`))
+                    entries.map(async ([key, value]) => execAsync(`setx ${key} ${quote(value)}`)),
                 );
             }
     }
@@ -83,15 +84,15 @@ async function getKeys(keyVaultClient, rc, vaultName) {
     const p = [];
     for (const secret of secretList) {
         if (secret.attributes.enabled) {
-            const secretName = secret.id.split('/').pop();
+            const secretName = secret.id.split("/").pop();
             // exclude secrets with automation prefix, which should only be used in automation
             if (!secretName.startsWith("automation")) {
                 p.push((async () => {
                     const response = await keyVaultClient.getSecret(vaultName, secretName);
-                    const envName = secretName.split('-').join('__'); // secret name can't contain underscores
+                    const envName = secretName.split("-").join("__"); // secret name can't contain underscores
                     console.log(`  ${envName}`);
                     rc.secrets[envName] = response.value;
-                })());;
+                })());
             }
         }
     }
@@ -101,7 +102,7 @@ async function getKeys(keyVaultClient, rc, vaultName) {
 
 async function execAsync(command, options) {
     return new Promise((res, rej) => {
-        exec(command, options, (err, stdout, stderr) => {
+        child_process.exec(command, options, (err, stdout, stderr) => {
             if (err) {
                 rej(err);
                 return;
@@ -116,7 +117,6 @@ async function execAsync(command, options) {
 
 class AzCliKeyVaultClient {
     static async get() {
-
         await execAsync("az ad signed-in-user show");
         return new AzCliKeyVaultClient();
 
@@ -131,22 +131,24 @@ class AzCliKeyVaultClient {
     }
 
     async getSecrets(vaultName) {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-return
         return JSON.parse(await execAsync(`az keyvault secret list --vault-name ${vaultName}`));
     }
 
     async getSecret(vaultName, secretName) {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-return
         return JSON.parse(await execAsync(`az keyvault secret show --vault-name ${vaultName} --name ${secretName}`));
     }
-};
+}
 
 class MsRestAzureKeyVaultClinet {
     static async get() {
-        const credentials = await msRestAzure.interactiveLogin();
+        const credentials = await interactiveLogin();
         return new MsRestAzureKeyVaultClinet(credentials);
     }
 
     constructor(credentials) {
-        this.client = new keyVault.KeyVaultClient(credentials);
+        this.client = new KeyVaultClient(credentials);
     }
 
     async getSecrets(vaultName) {
@@ -154,16 +156,16 @@ class MsRestAzureKeyVaultClinet {
     }
 
     async getSecret(vaultName, secretName) {
-        return this.client.getSecret(`https://${vaultName}.vault.azure.net/`, secretName, '');
+        return this.client.getSecret(`https://${vaultName}.vault.azure.net/`, secretName, "");
     }
 }
 
 async function getClient() {
-    return await AzCliKeyVaultClient.get();
+    return AzCliKeyVaultClient.get();
 
     // Disabling fallback to REST client while we decide how to streamline the getkeys tool
 
-    // const primary = await AzCliKeyVaultClient.get();
+    // const primary = AzCliKeyVaultClient.get();
     // if (primary !== undefined) {
     //     console.log("Using Azure CLI");
     //     return primary;
@@ -172,35 +174,36 @@ async function getClient() {
 }
 
 (async () => {
-    const rc = await rcTools.loadRC();
+    const rc = await loadRC();
 
     if (rc.secrets === undefined) {
         rc.secrets = {};
     }
 
-    // For debugging, change the following to 'false' to skip connecting to
+    // For debugging, change the following to 'false' and uncommented the if block to skip connecting to
     // Azure Key Vault and instead use the secrets cached in '~/.fluidtoolrc'.
-    if (true) {
-        const client = await getClient();
+    // if (true) {
+    const client = await getClient();
 
-        // Primary key vault for test/dev secrets shared by Microsoft-internal teams working on FF
-        await getKeys(client, rc, "prague-key-vault");
+    // Primary key vault for test/dev secrets shared by Microsoft-internal teams working on FF
+    await getKeys(client, rc, "prague-key-vault");
 
-        try {
-            // Key Vault with restricted access for the FF dev team only
-            await getKeys(client, rc, "ff-internal-dev-secrets");
-            console.log("\nNote: Default dev/test secrets overwritten with values from internal key vault.");
-        } catch (e) { }
-    }
+    try {
+        // Key Vault with restricted access for the FF dev team only
+        await getKeys(client, rc, "ff-internal-dev-secrets");
+        console.log("\nNote: Default dev/test secrets overwritten with values from internal key vault.");
+    } catch (e) { }
+    // }
 
     console.log(`\nWriting '${path.join(os.homedir(), ".fluidtoolrc")}'.`);
-    await rcTools.saveRC(rc);
+    await saveRC(rc);
     await saveEnv(rc.secrets);
 
-    console.warn(`\nFor the new environment to take effect, please restart your terminal.\n`)
-})().catch(e => {
+    console.warn(`\nFor the new environment to take effect, please restart your terminal.\n`);
+})().catch((e) => {
     if (e.message.includes("'az' is not recognized as an internal or external command")) {
         console.error(`ERROR: Azure CLI is not installed. Install it and run 'az login' before running this tool.`);
+        // eslint-disable-next-line no-undef
         exit(0);
     }
 

--- a/tools/getkeys/package.json
+++ b/tools/getkeys/package.json
@@ -11,8 +11,8 @@
   },
   "license": "MIT",
   "author": "Microsoft and contributors",
-  "main": "index.js",
   "type": "module",
+  "main": "index.js",
   "scripts": {
     "eslint": "eslint ./index.js",
     "eslint:fix": "eslint ./index.js --fix",

--- a/tools/getkeys/package.json
+++ b/tools/getkeys/package.json
@@ -12,7 +12,12 @@
   "license": "MIT",
   "author": "Microsoft and contributors",
   "main": "index.js",
+  "type": "module",
   "scripts": {
+    "eslint": "eslint ./index.js",
+    "eslint:fix": "eslint ./index.js --fix",
+    "lint": "npm run eslint",
+    "lint:fix": "npm run eslint:fix",
     "start": "node ./index.js",
     "test": "echo \"Error: no test specified\" && exit 1"
   },
@@ -20,5 +25,10 @@
     "@fluidframework/tool-utils": "^0.35.0",
     "azure-keyvault": "^3.0.4",
     "ms-rest-azure": "^2.6.0"
+  },
+  "devDependencies": {
+    "@fluidframework/eslint-config-fluid": "^1.0.0",
+    "eslint": "~8.6.0",
+    "typescript": "~4.5.5"
   }
 }

--- a/tools/getkeys/tsconfig.json
+++ b/tools/getkeys/tsconfig.json
@@ -1,0 +1,13 @@
+/*!
+* Adding this file to kick off exec lint script for js-only package
+*/
+{
+    "compilerOptions": {
+        "strictNullChecks": true,
+
+    },
+    "include": [
+      "**/*.js"
+    ],
+    "exclude": ["node_modules"]
+}

